bson-adt-mongo3-async/src/main/scala/adt/bson/mongo/FunctionImplicits.scala

```diff
@@ -0,0 +1,51 @@
+package adt.bson.mongo
+
+import com.mongodb.async.SingleResultCallback
+import play.api.libs.functional.ContravariantFunctor
+
+import scala.concurrent.Promise
+import scala.language.implicitConversions
+import scala.util.{Failure, Success, Try}
+
+// TODO: Avoid this redundant trait by moving the common mongo-core-driver dependent code to a shared module
+object FunctionImplicits extends FunctionImplicits
+trait FunctionImplicits {
+
+  implicit def asMongoFunction[P, R](f: P => R): com.mongodb.Function[P, R] = new MongoFunction(f)
+
+  implicit def asBlock[R](block: R => Any): com.mongodb.Block[R] = new MongoBlock(block)
+}
+
+class MongoFunction[P, R](f: P => R) extends com.mongodb.Function[P, R] {
+  override def apply(p: P): R = f(p)
+}
+
+class MongoBlock[P](f: P => Any) extends com.mongodb.Block[P] {
+  override def apply(p: P): Unit = f(p)
+}
+
+object MongoSingleCallback {
```

Comment:
I don't really care for the word "Single" in the name, but it mimics what MongoDB has.